src/machine.rs

```diff
@@ -176,11 +176,13 @@ impl MemoryExtra {
     ) {
         let ptr = ptr.assert_ptr();
         assert_eq!(ptr.offset, Size::ZERO);
-        this.memory
-            .extra
-            .extern_statics
-            .insert(Symbol::intern(name), ptr.alloc_id)
-            .unwrap_none();
+        assert_eq!(
+            this.memory
+                .extra
+                .extern_statics
+                .insert(Symbol::intern(name), ptr.alloc_id),
```

Comment:
I think it's bad style for assertions to have important side-effects like this. Can you let-bin the result of `insert`?

Reply:
Okay I should have just said "and the same everywhere else", sorry for that... but when I realized it affects basically all the lines you changed, I was already half-way through...

Reply:
Sure, I can change it. I'd be curious to hear why you think this is bad style though. Assertions with side effects are quite common in `rustc`. In C and C++ it's to be avoided because it's only conditionally compiled, but in Rust it's not much different than a `panic!()` in an `if`.

Reply:
I answered that in https://github.com/rust-lang/rust/issues/62633#issuecomment-790682732.